doc/readme.md

```diff
@@ -0,0 +1,67 @@
+Anecdotely it seems these two otherwise semantically identical pieces of code can behave differently because of JIT optimization decisions (x86):
+1. Constants in array
+```C#
+var a = new double[2] = { -2.5e160, 1.2e160 };
+double d = System.Math.Sqrt(a[0] * a[0] + a[1] * a[1]);
+System.Console.WriteLine("" + d); // Debug Build: Infinity, Release Build: 2.86356e160.
+```
+2. Constants in local variables
+```C#
+double a = -2.6e160;
+double b = -1.2e160;
+double d = System.Math.Sqrt(a * a + b * b);
+System.Console.WriteLine("" + d); // Debug Build: Infinity, Release Build: Infinity.
+```
+You would expect the result to always be Infinity as the intermediate value inside the Sqrt (8.2e320) is outside the valid double precision range. But depending on the platform, whether optimizations are enabled and surprisingly even whether or not the constants come from a local variable or array, this might produce a finite value. Seemingly this happens because it might decide to perform the calculation in extended precision.
+This behavior is specified in the C# specification (9.3.7) where it states: "Floating-point operations may be performed with higher precision than the result type of the operation.".
+Due to this inconsistency, we have to avoid floating point overflow behavior, as it is simple not guaranteed even using Microsoft's own runtime!
+
+Signed zeros
+-------------
+
+According to the C# specification section 9.3.7 C# supports and distinguishes between positive and negative floating point zeros. In practice we have found this not to be the case on our current Linux Mono environment, where seemingly negative zeroes are initialized as positive zeros. We should investigate this further at some point, but so far the affected tests are simply disabled in Linux using the `[WindowsOnlyAttribute]`.
+
+Known issues
+------------
+
+1. In order for code to be burstable we have to define all class-like objects as structs, but unlike classes there is no way to enforce that a struct gets initialized in any particular way. Just declaring a struct will give the user a zero-initialized instance. One instance where this limitation is very apparent is with `Random`. There is no way for us to guarantee that the user doesn't accidentally declare a zero-initialized instance. Such an instance has the unfortunate property that it will be stuck perpetually generating nothing but zeros. This is by the nature of the xor-shift algorithm, but it seems unnecessarily error-prone. While we could work around the problem by internally remapping the seed space, it would incur some unnecessary additional overhead and it seems this problem is likely to arise in other class-like objects implemented with structs.
```

Comment:
@dale did you address this with a PR that makes the seed be some function of the bit pattern, such that a 0 bit pattern isn't interpreted as a 0 seed value?

Reply:
I did add a function so you can give an index so that passing in zero is OK by processing the index through Wang hash, but it's still possible that someone could pick a value so that it hashes to zero and that will be used as the seed.

It still remains the case that if the internal `Random` state is zero, you get stuck in a bad state and can't generate anything else but zero.